File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        
        StringBuilder sb= new StringBuilder();
        sb.append(s.charAt(0));
        int count=1;
        for(int i=1;i<s.length();i++){
            if(s.charAt(i-1)==s.charAt(i) && count<2){
                count++;
                sb.append(s.charAt(i));
            }
            else if(s.charAt(i-1)!=s.charAt(i)){
                count=1;
                sb.append(s.charAt(i));
            }
        }
        return sb.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string result(1,s[0]);

        int count=1;
        for(int i=1;i<s.length();i++){
            if(s[i]==s[i-1] && count<2){
                count++;
                result.push_back(s[i]);
            } 
            else if(s[i]!=s[i-1]){
                count=1;
                result.push_back(s[i]);
            }
        }
        return result;
    }
};